front/src/modules/trainschedule/components/ManageTrainSchedule/Allowances/helpers.ts

```diff
@@ -31,15 +34,15 @@ export function findAllowanceOverlap({
     : [-1, -1];
 }
 
-export default function getAllowanceValue(value?: AllowanceValue) {
-  switch (value?.value_type) {
+export default function getAllowanceValue(value: AllowanceValueForm) {
```

Comment:
why do you want to introduce potentially undefined values and the need to deal with them everytime it,s called ?
the 0 returned here **is** the placeholder